src/components/Calculator.js

```diff
@@ -1,86 +1,212 @@
 import React, { Component } from 'react';
+import calculate from '../logic/calculate';
 
 // eslint-disable-next-line react/prefer-stateless-function
 export default class Calculator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    this.setState((state) => calculate(state, e.target.textContent));
+  }
+
   render() {
+    const { next, total } = this.state;
     return (
       <div className="flex justify-center items-center h-screen">
         <div className="w-2/6 shadow-lg">
           <div className="text-white bg-gray-500 flex justify-end p-5 w-full">
-            <span>0</span>
+            <span>{next || total || 0}</span>
           </div>
           <table className="w-full">
             <tbody className="w-2/6">
               <tr>
-                <td className="p-6 bg-white border hover:bg-gray-200 text-center">
-                  AC
-                </td>
-                <td className="p-6 bg-white border hover:bg-gray-200 text-center">
-                  +/-
-                </td>
-                <td className="p-6 bg-white border hover:bg-gray-200 text-center">
-                  %
-                </td>
-                <td className="p-6 border hover:bg-gray-200 text-center bg-orange-500 text-xl">
-                  ÷
+                <td className="bg-white border hover:bg-gray-200 text-center">
+                  <button
+                    className="p-6"
+                    type="button"
+                    onClick={this.handleClick}
+                  >
+                    AC
+                  </button>
+                </td>
+                <td className="bg-white border hover:bg-gray-200 text-center">
+                  <button
+                    className="p-6"
+                    type="button"
+                    onClick={this.handleClick}
+                  >
+                    +/-
+                  </button>
+                </td>
+                <td className="bg-white border hover:bg-gray-200 text-center">
+                  <button
+                    className="p-6"
+                    type="button"
+                    onClick={this.handleClick}
+                  >
```

Comment:
Please create a child component of the `Calculator` component and pass `handleClick` as props to it.

Reply:
Hello Sajjad, creating a child component for the Calculator that is empty is kind of getting me, confused bro. Don't know exactly what that component would do. What do you think?

Reply:
I'm having a single Calculator component and all the logic is working as expected.

Reply:
Hi, You could make another component named `Button`, put all the logic of button there and pass `handleClick` event as props to that child component.